src/razor/src/completion/razorCompletionItemProvider.ts

```diff
@@ -278,6 +287,38 @@ export class RazorCompletionItemProvider extends RazorLanguageFeatureBase implem
 
         return item;
     }
+
+    private static addUsingKeyword(
+        language: LanguageKind,
+        razorDocument: vscode.TextDocument,
+        hostDocumentPosition: vscode.Position,
+        completionItems: vscode.CompletionItem[]
+    ) {
+        // This is an ugly hack, but it's needed to get the "using" keyword to show up in the completion list.
+        // The reason it doesn't show up is because the C# generated document puts the position of the cursor
+        // at '__o = [||]', which isn't a valid location for a using statement.
+        if (language == LanguageKind.CSharp) {
+            const line = razorDocument.lineAt(hostDocumentPosition.line);
+            const lineText = line.text.substring(0, hostDocumentPosition.character);
+            if (
+                lineText.endsWith('@') ||
+                lineText.endsWith(
+                    '@u' ||
+                        lineText.endsWith('@us') ||
+                        lineText.endsWith('@usi') ||
+                        lineText.endsWith('@usin') ||
+                        lineText.endsWith('@using')
```

Comment:
You may have come up with the best way. If the user has already finished typing `@using`, does the item still need to be added?

```suggestion
                lineText.endsWith('@') ||
                    lineText.endsWith('@u') ||
                    lineText.endsWith('@us') ||
                    lineText.endsWith('@usi') ||
                    lineText.endsWith('@usin') ||
                    lineText.endsWith('@using')
```

Reply:
I considered this initially but wanted to match C#'s behavior. They seem to provide the completion item even if the user has finished typing out the whole word:
![image](https://github.com/dotnet/vscode-csharp/assets/16968319/fb249fde-b7b8-4490-852c-a9856b864bd3)